app/models/certificate-reference.ts

```diff
@@ -0,0 +1,21 @@
+import { Model, Prop, Record } from "app/core";
+
+export interface CertificateReferenceAttributes {
+    thumbprint: string;
+    thumbprintAlgorithm: string;
+    storeLocation: string;
+    storeName: string;
+    visibility: string[];
+}
+
+/**
+ * A reference to an application package to be deployed to a compute nodes
+ */
+@Model()
+export class CertificateReference extends Record<CertificateReferenceAttributes> {
+    @Prop() public thumbprint: string;
+    @Prop() public thumbprintAlgorithm: string;
+    @Prop() public storeLocation: string;
+    @Prop() public storeName: string;
+    @Prop() public visibility: string[];
```

Comment:
Needs to be ListProp here

```
@ListProp(String) public visibility: List<string> = List([]);
```

Don't forget to set the default value to be an empty list too that way we never have to check for null for list